### PR TITLE
Template: add option to exclude code linters for custom pipeline template

### PR DESCRIPTION
**Pull request by mirpedrol**
Part of https://github.com/nf-core/tools/issues/2340 and https://github.com/nf-core/tools/issues/2342

### CODE REVIEW

File: tests/test_create.py [hunk 1]
```diff
@@ -106,14 +106,16 @@ def test_pipeline_creation_with_yml_skip(self, tmp_path):
         pipeline.init_pipeline()
 
         # Check pipeline template yml has been dumped to `.nf-core.yml` and matches input
-        assert not os.path.exists(os.path.join(pipeline.outdir, "pipeline_template.yml"))
-        assert os.path.exists(os.path.join(pipeline.outdir, ".nf-core.yml"))
-        with open(os.path.join(pipeline.outdir, ".nf-core.yml")) as fh:
+        assert not os.path.exists(Path(pipeline.outdir / "pipeline_template.yml"))
+        assert os.path.exists(Path(pipeline.outdir / ".nf-core.yml"))
```
Comment (mashehu) [inline tests/test_create.py:110]:
```suggestion
        assert not Path(pipeline.outdir / "pipeline_template.yml").exists()
        assert Path(pipeline.outdir / ".nf-core.yml").exists()
```
same below

**Review by mashehu — APPROVED** [review]

File: tests/test_create.py [hunk 2]
```diff
@@ -106,14 +106,16 @@ def test_pipeline_creation_with_yml_skip(self, tmp_path):
...
+        with open(Path(pipeline.outdir / ".nf-core.yml")) as fh:
```
Comment (mirpedrol) [inline tests/test_create.py:111]:
```suggestion
        assert not os.path.exists(pipeline.outdir / "pipeline_template.yml")
        assert os.path.exists(pipeline.outdir / ".nf-core.yml")
        with open(pipeline.outdir / ".nf-core.yml") as fh:
```

File: tests/test_create.py [hunk 3]
```diff
@@ -106,14 +106,16 @@ def test_pipeline_creation_with_yml_skip(self, tmp_path):
...
             assert yaml.safe_load(PIPELINE_TEMPLATE_YML.read_text()).items() <= nfcore_yml["template"].items()
 
         # Check that some of the skipped files are not present
-        assert not os.path.exists(os.path.join(pipeline.outdir, "CODE_OF_CONDUCT.md"))
-        assert not os.path.exists(os.path.join(pipeline.outdir, ".github"))
-        assert not os.path.exists(os.path.join(pipeline.outdir, "conf", "igenomes.config"))
+        assert not os.path.exists(Path(pipeline.outdir / "CODE_OF_CONDUCT.md"))
+        assert not os.path.exists(Path(pipeline.outdir / ".github"))
+        assert not os.path.exists(Path(pipeline.outdir / "conf" / "igenomes.config"))
+        assert not os.path.exists(Path(pipeline.outdir / ".editorconfig"))
```
Comment (mirpedrol) [inline tests/test_create.py:120]:
```suggestion
        assert not os.path.exists(pipeline.outdir / "CODE_OF_CONDUCT.md")
        assert not os.path.exists(pipeline.outdir / ".github")
        assert not os.path.exists(pipeline.outdir / "conf" / "igenomes.config")
        assert not os.path.exists(pipeline.outdir / ".editorconfig")
```

Comment (mashehu) [reply tests/test_create.py:120]: why are you remove path but not os.path.exists? 😁

Comment (mirpedrol) [reply tests/test_create.py:120]: I overlooked the `os.path.exists` 😂

**Pull request merged**